services/terrain/src/terrain/services/permanent_id_requests.clj

```diff
@@ -38,11 +44,21 @@
              :path staging-dest})))
   data-item)
 
+(defn- validate-publish-dest
+  [{:keys [path] :as data-item}]
+  (let [publish-dest (ft/path-join (config/permanent-id-publish-dir) (ft/basename path))]
+    (when (data-info/path-exists? (config/irods-user) publish-dest)
+      (throw+ {:type :clojure-commons.exception/exists
+               :error "A folder with this name has already been published."
+               :path publish-dest})))
```

Comment:
What I mean is, checking that the path exists and throwing a `clojure-commons.exception/exists` error is all the sort of stuff data-info is doing for all sorts of other things -- this is just `validators/path-not-exists`, basically. The only thing that's different is the more-specific error message about a folder with that name already having been published. Or, put differently, the link between a path existing when it shouldn't and the `clojure-commons.exception/exists` error type is the concern of data-info, despite the reason the path should not exist being unrelated.